pkg/systemd/generate/containers.go

```diff
@@ -349,6 +350,25 @@ func executeContainerTemplate(info *containerInfo, options entities.GenerateSyst
 
 		startCommand = append(startCommand, remainingCmd...)
 		startCommand = escapeSystemdArguments(startCommand)
+		if options.TemplateUnitFile {
+			info.IdentifySpecifier = true
+			nameIx := -1
+			for ix, arg := range startCommand {
+				if strings.HasPrefix(arg, "--name=") {
+					nameIx = ix
+					break
+				}
+				if arg == "--name" {
+					nameIx = ix + 1
+					break
+				}
+			}
+
+			if nameIx != -1 {
+				startCommand[nameIx] = fmt.Sprintf("%s-%%i", startCommand[nameIx])
+			}
```

Comment:
Why is that code needed at all? I didn't read much of the code yet.  This part here may deserve some code comments to leave breadcrumbs for future generations.

Reply:
@vrothberg and @Luap99 looks like it's my fault. According to the diff here https://github.com/containers/podman/issues/10275#issue-880513405 this is unnecessary. I am removing this change from the code. Thanks.

Reply:
I think this is required otherwise template is useless with a name set. If you want to start to containers with `systemctl start service@con1 service@con2` it will fail because podman can only create one container with this name. Maybe even worse because `--replace` is set it will overwrite the other container.

Reply:
Alright then I guess we should put some comments explaining why this is necessary as @vrothberg asked.
Updating the code...
Thanks @Luap99

Reply:
Still open.

Reply:
Probably worth creating a dedicated function along with unit tests to make sure we're never regressing on it.